source/particle/interpolator/bilinear_least_squares.cc

```diff
@@ -278,8 +279,8 @@ namespace aspect
       ASPECT_REGISTER_PARTICLE_INTERPOLATOR(BilinearLeastSquares,
                                             "bilinear least squares",
                                             "Interpolates particle properties onto a vector of points using a "
-                                            "bilinear least squares method. Currently only 2D models are "
-                                            "supported. Note that deal.II must be configured with BLAS/LAPACK.")
+                                            "bilinear least squares method. Note that deal.II must be configured"
```

Comment:
Now that you removed the LAPACKFullMatrix and the SVD call it should be possible to use this without LAPACK do I understand that correctly? If so please remove the note about BLAS/LAPACK here.

Reply:
@gassmoeller Someone removed this comment from the earlier version of bilinear_least_squares.cc :

>         // Matrix A can be rank deficient if it does not have full rank, therefore singular.
        // To circumvent this issue, we solve A^TAx=A^Tr by using singular value
        // decomposition (SVD).

Did you do this?  

@sac-bsa and I really want to put in the SVD algorithm, since it supposedly can handle a rank deficient matrix and would therefore make this part of the algorithm 'bullet proof'.   However we wouldn't solve the 'Normal Equations'  A^TAx=A^Tr , since that is the part that is wildly unstable. We would feed it A x = b (or r) and then the SVD would factor A = U \Sigma V^T, etc.

The current QR implementation should be much better / more accurate, which numerical linear algebra people refer to as more stable, by, in some cases many, many digits.  I can post or email an example to you. Let me know.

However, in the long run, I would like to implement the SVD algorithm for solving LS problems when A is rank deficient.  @sac-bsa and I both understand the theory, but we have to figure out whether that particular call to the SVD would return the minimizer in the L2 norm even when A is rank deficient.  The time factor depends on how much time Mack and my classes take out of our time for research budget.

Reply:
Internally the QR class has a LAPACKFullMatrix in which it stores R. From what I can tell, it uses BLAS as well for its multiplication routines, because of this, I believe that deal.ii would still need to be configured with BLAS/LAPACK.

Reply:
> Internally the QR class has a LAPACKFullMatrix in which it stores R. From what I can tell, it uses BLAS as well for its multiplication routines, because of this, I believe that deal.ii would still need to be configured with BLAS/LAPACK.

Ah, ok then leave the comment in. Still since the FullMatrix is not longer used directly in this file you can remove the fullmatrix include at the top of the file.

> Someone removed this comment from the earlier version of bilinear_least_squares.cc. Did you do this?

I dont think so (dont quite remember), but I approved the change because the sentence was worded a bit repetitively. I thought we kept the note that we use SVD because A might be rank deficient, but I do not see it in the current code. If you want to add a note that A might be rank deficient and what that means for the current algorithms I would be fine with it.